crates/adapters/src/controller/stats.rs

```diff
@@ -906,9 +872,11 @@ impl InputEndpointStatus {
         }
     }
 
-    fn consume_buffered(&self) {
+    pub(crate) fn consume_buffered(&self, num_records: u64) {
         self.metrics.buffered_bytes.store(0, Ordering::Release);
```

Comment:
I wasn't sure what to do about buffered bytes. If we want it to be accurate, we need to track the number of bytes that correspond to each individual record, which we don't do now. That's a lot more bookkeeping to thread through the system (we will need to pair every record with a byte count, I think), and it might be difficult to get it to be completely right and to test it confidently.

This current approach of reporting incorrect buffered bytes is clearly wrong (I forgot about that before I submitted this). One option would be to drop the buffered bytes counter and only report ingested bytes; that would be a pretty easy fix, I think.

I guess that we'll need to actually track the byte size of individual records if we want to support step size in terms of bytes instead of in terms of records. I don't really want to do that in this PR because I think it will add complication.

What are your thoughts?

Reply:
I am ok with not reporting buffered bytes for now and maybe adding bookkeeping in the future if there's demand for it.

Reply:
I added a commit that drops buffered byte tracking.